core/domain/exp_domain.py

```diff
@@ -974,53 +972,6 @@ def validate(self):
             trigger.customization_arg_specs)
 
 
-class Fallback(object):
-    """Value object representing a fallback.
-
-    A fallback consists of a trigger and an outcome. When the trigger is
-    satisfied, the user flow is rerouted to the given outcome.
-    """
-    def __init__(self, trigger, outcome):
-        """Initializes a Fallback domain object.
-
-        Args:
-            trigger: TriggerInstance. The satisfied trigger.
-            outcome: Outcome. The outcome to apply when the user hits the
-                trigger.
-        """
-        self.trigger = trigger
-        self.outcome = outcome
-
-    def to_dict(self):
-        """Returns a dict representing this Fallback domain object.
-
-        Returns:
-            dict. A dict mapping all fields of Fallback instance.
-        """
-        return {
-            'trigger': self.trigger.to_dict(),
-            'outcome': self.outcome.to_dict(),
-        }
-
-    @classmethod
-    def from_dict(cls, fallback_dict):
-        """Return a Fallback domain object from a dict.
-
-        Args:
-            fallback_dict: dict. The dict representation of Fallback object.
-
-        Returns:
-            Fallback. The corresponding Fallback domain object.
-        """
-        return cls(
-            TriggerInstance.from_dict(fallback_dict['trigger']),
```

Comment:
Is TriggerInstance still needed? If not, should we delete that (and all associated stuff) too?